modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -213,54 +213,43 @@ class SeqexecEngine(httpClient: Client[IO], settings: SeqexecEngine.Settings, sm
+        (EngineState.queues ^|-? index(qid)).modify(_.addSeq(seqId))(st)
       else st
     }
   ).getOrElse(st)
 
-  def addSequenceToQueue(q: EventQueue, queueName: String, seqId: Observation.Id): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
-    Event.modifyState[executeEngine.ConcreteTypes](addSeq(queueName, seqId) withEvent NullSeqEvent)
+  def addSequenceToQueue(q: EventQueue, qid: QueueId, seqId: Observation.Id): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
+    Event.modifyState[executeEngine.ConcreteTypes](addSeq(qid, seqId) withEvent NullSeqEvent)
   ).map(_.asRight)
 
-  private def removeSeq(name: String, seqId: Observation.Id): Endo[EngineState] = st => (
+  private def removeSeq(qid: QueueId, seqId: Observation.Id): Endo[EngineState] = st => (
     for {
-      q <- st.queues.get(name)
-    } yield if(!q.status(st).isRunning && q.contains(seqId))
-        (EngineState.queues ^|-? index(name)).modify(_.filterNot(_===seqId))(st)
+      q <- st.queues.get(qid)
+    } yield if(q.status(st) =!= SequencesBatch.State.Running && q.queue.contains(seqId))
+        (EngineState.queues ^|-? index(qid)).modify(_.removeSeq(seqId))(st)
       else st
   ).getOrElse(st)
 
-  def removeSequenceFromQueue(q: EventQueue, queueName: String, seqId: Observation.Id): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
-    Event.modifyState[executeEngine.ConcreteTypes](removeSeq(queueName, seqId) >>>{(_, NullSeqEvent)})
+  def removeSequenceFromQueue(q: EventQueue, qid: QueueId, seqId: Observation.Id): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
+    Event.modifyState[executeEngine.ConcreteTypes](removeSeq(qid, seqId) withEvent NullSeqEvent)
   ).map(_.asRight)
 
-  // This assumes that there is only one instance of e in l
-  private def moveElement[T](l: List[T], e: T, d: Int)(implicit eq: Eq[T]): List[T] = {
-    val idx = l.indexOf(e)
-
-    if(d === 0 || idx<0) l
-    else {
-      val (h, t) = l.filterNot(_ === e).splitAt(idx+d)
-      (h :+ e) ::: t
-    }
-  }
-
-  private def moveSeq(name: String, seqId: Observation.Id, d: Int): Endo[EngineState] = st => (
+  private def moveSeq(qid: QueueId, seqId: Observation.Id, d: Int): Endo[EngineState] = st => (
     for {
-      q <- st.queues.get(name)
-    } yield if(!q.status(st).isRunning && q.contains(seqId))
-        (EngineState.queues ^|-? index(name)).modify(moveElement(_, seqId, d))(st)
+      q <- st.queues.get(qid)
+    } yield if(q.status(st) =!= SequencesBatch.State.Running && q.queue.contains(seqId))
+        (EngineState.queues ^|-? index(qid)).modify(_.moveSeq(seqId, d))(st)
       else st
   ).getOrElse(st)
 
-  def moveSequenceInQueue(q: EventQueue, queueName: String, seqId: Observation.Id, d: Int): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
-    Event.modifyState[executeEngine.ConcreteTypes](moveSeq(queueName, seqId, d) withEvent NullSeqEvent)
+  def moveSequenceInQueue(q: EventQueue, qid: QueueId, seqId: Observation.Id, d: Int): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
```

Comment:
Should there be checks  on the range of d?

Reply:
That is not necessary. The element will be moved to the first or last position if `d` is out of range.